### PR TITLE
Make the ensemble size configurable

### LINKED ISSUE #1216: Make the ensemble size configurable
Currently the ensemble size is hard coded to 1000 in the code. This should be read from the `ensemble.yml` file instead.

### PR DESCRIPTION
**Pull request by kvashchuka**
**Issue**
Resolves #1216 

**Approach**
Read the ensemble size from `size` field in the ensemble.yml file

### CODE REVIEW

**Review by markusdregi — COMMENTED** [review]
Looks like you found the correct line to change ;)

A few tests would be good though 👍

**Comment by kvashchuka** [comment]
There is already `_assert_ensemble_size` function though. Which test do you have in mind? 
Also, what should happen if the ensemble size is not specified? Error message, right? Should we catch this situation during run or we will implement ensemble.yml file before-run check?

**Comment by markusdregi** [comment]
First, I'm thinking that proper validation of configuration is beyond the scope of this PR. I suggest that we just burn heavily with the `KeyError` currently being thrown if there is no `size` key...

But, I would really like to see a test that verifies that the value is read correctly for the happy path. The fact that you were able to change the implementation, but not any of the tests hints towards that the current test base is not sufficient. I have a suggestion for how to do this. Give me a few minutes 👍

File: examples/polynomial/evaluation/ensemble.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-size: 1000
+size: 1001
```
Comment (markusdregi) [inline examples/polynomial/evaluation/ensemble.yml:1]: Good 👏 Now we have at least two integration tests with different ensemble sizes!

**Review by markusdregi — APPROVED** [review]
Looks good 🚀

**Pull request merged**